frb_example/pure_dart/dart/lib/main.dart

```diff
@@ -692,7 +693,30 @@ void main(List<String> args) async {
     });
   });
 
-  group('opaque type', () {
+  group('dart opaque type', () {
```

Comment:
would be great to have tests for *ephemeral* and *large* objects to detect memory leak. For example, `Uint8List` with 200MB size.

I have some boilerplate code for ensuring there is no memory leak, and I can provide it here. Ping me if I do not paste it.

Reply:
ping

Reply:
Done. https://stackoverflow.com/a/74572599/4619958 (https://gist.github.com/fzyzcjy/e68c375643d7c77942cdc8fb5f01de18)

Reply:
Hmm may I know where is it implemented?

Reply:
is that not what you need?

```
    test('drop', () async {
      expect(await api.asyncAcceptDartOpaque(opaque: _createLargeList(mb: 200)), 'async test');
      expect(api.syncAcceptDartOpaque(opaque: _createLargeList(mb: 200)), 'test');
    });
 ```

Reply:
Well, we should monitor whether the list is GCed. See "usage_example" in https://gist.github.com/fzyzcjy/e68c375643d7c77942cdc8fb5f01de18#file-usage_example-dart-L80, especially https://gist.github.com/fzyzcjy/e68c375643d7c77942cdc8fb5f01de18#file-usage_example-dart-L88 where we check Weak ref is null

Reply:
Will this be enough?
   
   ```
    test('GC', () async {
      var vmService = await VmServiceUtil.create();
      Uint8List? strongRef = _createLargeList(mb: 300);
      final weakRef = WeakReference(strongRef);
      await api.setStaticDartOpaque(opaque: strongRef);
      strongRef = null;

      await vmService.gc();
      await Future<void>.delayed(const Duration(milliseconds: 10));
      expect(weakRef.target , isNotNull);

      await api.dropStaticDartOpaque();
      await vmService.gc();
      await Future<void>.delayed(const Duration(milliseconds: 10));
      expect(weakRef.target , isNull);

      vmService.dispose();
    });
  ```

Reply:
on the web platform: Unsupported operation: new RawReceivePort
on the valgring: Exception: Cannot find serverUri for VmService. Ensure you run like `dart run --enable-vm-service path/to/your/file.dart`

is it worth testing this particular case without valgrind?
(dart run --enable-vm-service locally this test worked)

Reply:
LGTM as long as tests pass. For a more robust test, maybe we can refer to my example and add a for loop to retry.

We may temporarily ignore web if it is too hard.

> is it worth testing this particular case without valgrind

IMHO yes